### PR TITLE
Bug 1881422: Issue a warning when two or more profiles use the same priority.

### PR DESCRIPTION
**Pull request by jmencak**
When two or more profiles have the same priority and they match a node, the
profile selection between these used to be random.  This PR adds some determinism
to the way profile selection is done based on Tuned CR names and also issues a
warning about this to help cluster administrators to discover this problem.

### CODE REVIEW

File: pkg/operator/profilecalculator.go [hunk 1]
```diff
@@ -464,6 +464,16 @@ func tunedRecommend(tunedSlice []*tunedv1.Tuned) []tunedv1.TunedRecommend {
 		return recommendAll[i].Priority != nil // undefined priority has the lowest priority
 	})
 
+	for i := 0; i < len(recommendAll)-1; i++ {
```
Comment (cynepco3hahue) [inline pkg/operator/profilecalculator.go:467]: Does `recommendAll` sorted by priority?
Like I can imagine the situation when you will have
[profile1, profile2, profile3] and
profile1.priority=30
profile2.priority=50
profile3.priority=30

Comment (jmencak) [reply pkg/operator/profilecalculator.go:467]: Yep, sorted by priority by the code above.

**Comment by jmencak** [comment]
Thanks for the review @cynepco3hahue .  After talking to Martin S. I've decided to add some determinism to this too.

**Pull request merged**